backend/canisters/user/impl/src/updates/set_pin_number.rs

```diff
@@ -20,30 +22,50 @@ fn set_pin_number(args: Args) -> Response {
 fn set_pin_number_impl(args: Args, state: &mut RuntimeState) -> Response {
     let now = state.env.now();
 
-    if let Err(error) = state.data.pin_number.verify(args.current.as_deref(), now) {
-        match error {
-            VerifyPinError::PinRequired => PinRequired,
-            VerifyPinError::PinIncorrect(delay) => PinIncorrect(delay),
-            VerifyPinError::TooManyFailedAttempted(delay) => TooManyFailedPinAttempts(delay),
-        }
-    } else {
-        if let Some(new) = args.new.as_ref() {
-            let length = new.len();
-            if length < MIN_LENGTH {
-                return TooShort(FieldTooShortResult {
-                    length_provided: length as u32,
-                    min_length: MIN_LENGTH as u32,
-                });
+    if state.data.pin_number.enabled() {
+        match args.verification {
+            PinNumberVerification::None => return PinRequired,
```

Comment:
Why does this variant exist?

Reply:
If you set it for the first time you don't need an existing pin